networking/ipv6.md

```diff
@@ -1,217 +1,159 @@
+1. Edit the CNI config (`calico-config` ConfigMap in the manifest), and enable IPv4 and IPv6 address allocation by setting both fields to true.
 
-These requirements are not yet all met in common cloud images—but it
-is easy to remedy that by launching an image, making appropriate changes
-to its configuration files, taking a snapshot, and then using that
-snapshot thereafter instead of the original image.
-
-For example, starting from an Ubuntu cloud image, the following
-changes will suffice to meet the requirements just listed.
-
--   In `/etc/network/interfaces.d/eth0.cfg`, add:
-
-        iface eth0 inet6 dhcp
-                accept_ra 1
-
--   In `/sbin/dhclient-script`, add at the start of the script:
-
-        new_ip6_prefixlen=128
+   ```
+       "ipam": {
+           "type": "calico-ipam",
+           "assign_ipv4": "true",
+           "assign_ipv6": "true"
+       },
+   ```
 
--   In `/etc/sysctl.d`, create a file named `30-eth0-rs-delay.conf` with
-    contents:
+1. Configure IPv6 support and the default IPv6 IP pool by adding the following variable settings to the environment for the `calico-node` container:
 
-        net.ipv6.conf.eth0.router_solicitation_delay = 10
+   | Variable name | Value |
+| ------------- | ----- |
+| `IP6`         | `autodetect` |
+| `CALICO_IPV6POOL_CIDR` | The same as the IPv6 range you configured as the cluster CIDR to kube-controller-manager and kube-proxy |
+| `FELIX_IPV6SUPPORT` | `true` |
 
-For CentOS, these additions to a cloud-init script have been reported to be effective:
+1. Apply the edited manifest with `kubectl apply -f`. 
+ 
+   New pods will get IPv6 addresses as well as IPv4, and can communicate with each other and the outside world over IPv6.
 
-	runcmd:
-	- sed -i -e '$a'"IPV6INIT=yes" /etc/sysconfig/network-scripts/ifcfg-eth0
-	- sed -i -e '$a'"DHCPV6C=yes" /etc/sysconfig/network-scripts/ifcfg-eth0
-	- sed -i '/PATH/i\new_ip6_prefixlen=128' /sbin/dhclient-script
-	- systemctl restart network
+%>
+{% endtabs %}
 
 
-### Enabling IPv6 support in {{site.prodname}}
+#### Change host IPv4 addresses to IPv6 only
 
-In OpenStack, IPv6 connectivity requires defining an IPv6 subnet, in
-each Neutron network, with:
+To switch the host to use IPv6 only, follow these additional steps.
 
--   the IPv6 address range that you want your VMs to use
--   DHCP enabled
--   IPv6 address mode set to DHCPv6 stateful.
+1. Disable [IP autodetection of IPv4]({{site.baseurl}}//networking/ip-autodetection) by setting `IP` to `none`.
+1. Calculate the {{site.prodname}} BGP router ID for IPv6 using either of the following methods.
+   - Set the environment variable `CALICO_ROUTER_ID=hash` on {{site.nodecontainer}}.     
+     This configures {{site.prodname}} to calculate the router ID based on the hostname.
+   - Pass a unique value for `CALICO_ROUTER_ID` to each node individually. 
```

Comment:
The detail here is fine, but I'm afraid the positioning and narrative of this section (Change host IPv4 addresses to IPv6 only) is not quite right.

It should be an optional sub-section **within** "Enable IPv6 only".  The narrative should be (but obviously please reword as you think fit): "If you want your workloads to have IPv6 addresses only, and the reason for that is because you don't have IPv4 addresses or connectivity between your **nodes**, you need the following additional steps to tell {{site.prodname}} not to look for any IPv4 addresses."

It makes sense only within "Enable IPv6 only", and **not** within "Enable dual stack", because dual stack **requires** the nodes to have IPv4 addresses and connectivity.

Reply:
fixed.